### PR TITLE
Create known-issues.md

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

Bringing across the known issues from the old k8s section of Fermyon Developer Home. This page can be updated with known issues on an ongoing basis. Please feel free to remove any issues that are no longer relevant.

### CODE REVIEW

File: content/en/docs/spin-operator/support/known-issues.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+[...]
+[component.static-file-server]
+source = "./spin_static_fs.wasm"
+files = [{ source = "./", destination = "/" }]
+[...]
+```
+
+A workaround is to mount specific files or directories that don't already exist in the container.
+
+### Limited Volume Mounts
```
[the remark this answers is not in the document]
Comment (macolso) [reply content/en/docs/spin-operator/support/known-issues.md:49]: Yeah, I don't think we mount volumes for SpinApps the way you would for pods so we can close remove this

File: content/en/docs/spin-operator/support/known-issues.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+Currently, the spin shim launches Spin with pre-defined settings and does not support custom configurations.
+
+This means you cannot set environment variables or use certain `spin up` flags, like `--disable-cache`, `--direct-mounts`, or `--state-dir`.
+
+The preset values include:
+
+- `--listen: 0.0.0.0:80`
+- `--runtime-config-file: /runtime-config.toml`
+
+Additionally, `spin.toml` should be located at `/spin.toml` in the container. The same applies to `runtime-config.toml`, which should be at `/runtime-config.toml`. If there's no runtime-config file, the shim will ignore it.
```
Comment (macolso) [inline content/en/docs/spin-operator/support/known-issues.md:27]: This statement no longer applies

File: content/en/docs/spin-operator/support/known-issues.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+To set up Spin on Kubernetes, we use the [Containerd-Shim for Spin](https://github.com/deislabs/containerd-wasm-shims), a great contribution from the community. While Spin mostly works as expected, there are a few important points to note.
+
+The project is still evolving, and we anticipate enhancements with each new release of containerd-spin-shim.
+
+- [Specific Configuration of `spin up`](#specific-configuration-of-spin-up)
+- [Limitation in File Mounts for Components](#limitation-in-file-mounts-for-components)
+- [Limited Volume Mounts](#limited-volume-mounts)
+
+### Specific Configuration of `spin up`
```
[the remark this answers is not in the document]
Comment (macolso) [reply content/en/docs/spin-operator/support/known-issues.md:16]: I don't think this applies - how would spin up work for a SpinApp running on a remote cluster (or locally on k3d). Isn't it run via a deployment on k8s or locally directly with Spin as a Spin App?

File: content/en/docs/spin-operator/support/known-issues.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+- `--listen: 0.0.0.0:80`
+- `--runtime-config-file: /runtime-config.toml`
+
+Additionally, `spin.toml` should be located at `/spin.toml` in the container. The same applies to `runtime-config.toml`, which should be at `/runtime-config.toml`. If there's no runtime-config file, the shim will ignore it.
+
+For now, if you need a specific setup, your only option is to build your version of `containerd-spin-shim`.
+
+Keep track of this [open issue](https://github.com/deislabs/containerd-wasm-shims/issues/166) for updates on configuration injection into the shim.
+
+### Limitation in File Mounts for Components
```
[the remark this answers is not in the document]
Comment (macolso) [reply content/en/docs/spin-operator/support/known-issues.md:33]: @calebschoepp - do you know if we support file mounts for SpinApps in `spin.toml`?

[the remark this answers is not in the document]
Comment (calebschoepp) [reply content/en/docs/spin-operator/support/known-issues.md:33]: I haven't personally tested it, but I think we should be able to use `volumes` and `volumeMounts` to make sure that Spin has access to the necessary files.

**Comment by bacongobbler** [comment]
closing as stale.

**Pull request closed without merging**